dependencyManagement/build.gradle.kts

```diff
@@ -2,96 +2,60 @@ plugins {
-  "com.google.errorprone:error_prone_annotations:${errorProneVersion}",
-  "com.google.errorprone:error_prone_core:${errorProneVersion}",
-  "io.github.netmikey.logunit:logunit-jul:2.0.0",
-  "io.opentelemetry.proto:opentelemetry-proto:1.5.0-alpha",
-  // these two constraints can be removed once the opentelemetry-instrumentation-bom-alpha
-  // is updated to contain the latest version of opentelemetry-semconv
-  "io.opentelemetry.semconv:opentelemetry-semconv:${semConvVersion}",
-  "io.opentelemetry.semconv:opentelemetry-semconv-incubating:${semConvVersion}",
-  "io.prometheus:simpleclient:${prometheusVersion}",
-  "io.prometheus:simpleclient_common:${prometheusVersion}",
-  "io.prometheus:simpleclient_httpserver:${prometheusVersion}",
-  "org.mockito:mockito-core:${mockitoVersion}",
-  "org.mockito:mockito-inline:${mockitoVersion}",
-  "org.mockito:mockito-junit-jupiter:${mockitoVersion}",
-  "org.slf4j:slf4j-api:${slf4jVersion}",
-  "org.slf4j:slf4j-simple:${slf4jVersion}",
-  "org.slf4j:log4j-over-slf4j:${slf4jVersion}",
-  "org.slf4j:jcl-over-slf4j:${slf4jVersion}",
-  "org.slf4j:jul-to-slf4j:${slf4jVersion}"
-)
-
-val DEPENDENCIES = listOf(
-  "com.google.code.findbugs:annotations:3.0.1u2",
-  "com.google.code.findbugs:jsr305:3.0.2",
-  "com.squareup.okhttp3:okhttp:4.12.0",
-  "com.uber.nullaway:nullaway:0.12.3",
-  "org.assertj:assertj-core:3.27.3",
-  "org.awaitility:awaitility:4.2.2",
-  "org.bouncycastle:bcpkix-jdk15on:1.70",
-  "org.junit-pioneer:junit-pioneer:1.9.1",
-  "org.skyscreamer:jsonassert:1.5.3",
-  "org.apache.kafka:kafka-clients:3.9.0",
-  "org.testcontainers:kafka:1.20.4",
-  "com.lmax:disruptor:3.4.4",
-  "org.jctools:jctools-core:4.0.5",
-  "tools.profiler:async-profiler:3.0",
-  "com.blogspot.mydailyjava:weak-lock-free:0.18",
-  "org.agrona:agrona:1.22.0"
-)
+val semconvVersion = "1.30.0-rc.1"
 
 javaPlatform {
   allowDependencies()
 }
 
 dependencies {
-  for (bom in DEPENDENCY_BOMS) {
-    api(enforcedPlatform(bom))
-    val split = bom.split(':')
-    dependencyVersions[split[0]] = split[2]
-  }
+  // boms that are only used by tests should be added in otel.java-conventions.gradle.kts
+  // under JvmTestSuite so they don't show up as runtime dependencies in license and vulnerability scans
+  // (the constraints section below doesn't have this issue, and will only show up
+  // as runtime dependencies if they are actually used as runtime dependencies)
+  api(enforcedPlatform("io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:${otelInstrumentationVersion}"))
+  api(enforcedPlatform("com.fasterxml.jackson:jackson-bom:2.18.2"))
+
   constraints {
-    for (dependency in CORE_DEPENDENCIES) {
-      api(dependency)
-      val split = dependency.split(':')
-      dependencyVersions[split[0]] = split[2]
-    }
-    for (dependency in DEPENDENCIES) {
-      api(dependency)
-      val split = dependency.split(':')
-      dependencyVersions[split[0]] = split[2]
-    }
```

Comment:
getting rid of this was unrelated, but we don't use the `dependencyVersions` constants, and inlining the dependencies directly into this section seems so much more straightforward